PlaybackLab/ControlsLockScreenSampleApp/ControlsLockScreenSampleApp/Views/PlayerViewController.swift

```diff
@@ -9,23 +9,55 @@ import UIKit
 import AVFoundation
 import MediaPlayer
 
-class PlayerViewController: OOOoyalaPlayerViewController {
+//now inited from I.B.
+class PlayerViewController: UIViewController {
   
+  // MARK: - Private properties
   // properties for the video
   public var option: PlayerSelectionOption!
   
   // properties required for a Fairplay asset
   private var apiKey: String?
   private var apiSecret: String?
+  private var ooyalaPlayerVC: OOOoyalaPlayerViewController!
   
   // remote control center
   private let remoteCommandCenter = MPRemoteCommandCenter.shared()
   
-  override func viewWillAppear(_ animated: Bool) {
-    super.viewWillAppear(animated)
+  // MARK: - View controller lifecycle
+  override func viewDidLoad() {
+    super.viewDidLoad()
     
     self.title = option.title
     
+    setupOoyalaPlayerStuff()
+    
+    addPlayerViewController(vc:ooyalaPlayerVC)
+    
+    addObservers()
+    setupCommandCenter()
+    setupPlayingInfoCenter()
+  }
+  
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    
+    // Load the video, step 2/2
+    ooyalaPlayerVC.player.play()
+  }
+  
+  // MARK: - Private methods
+  private func addPlayerViewController(vc: UIViewController) {
+    
+    vc.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    self.addChild(vc)
+    view.addSubview(vc.view)
+  }
+  
+  private func setupOoyalaPlayerStuff() {
+    
+    var player: OOOoyalaPlayer!
```

Comment:
Does it really need to be an implicit unwrapped optional?

Reply:
+